crossai/loader/_audio.py

```diff
@@ -88,17 +88,19 @@ def audio_loader(path, sr=22500, n_workers=min(mp.cpu_count(), 4), norm_range =
     progress.update(1)
     progress.set_description("Load data into a dataframe")
 
+    df = data
+
     for i in range(len(data)):
-        for j in range(len(data[i])):
-            data[i][j] = (data[i][j].astype(np.float32), subdirnames[i])
+            for j in range(len(data[i])):
+                data[i][j] = (data[i][j], subdirnames[i])
 
-    df = pd.DataFrame(columns=['data', 'label', 'indice'])
+    df = pd.DataFrame(columns=['data', 'label', 'filename'])
 
     for i in range(len(data)):
-        for j in range(len(data[i])):
-            df.loc[len(df)] = [data[i][j][0], data[i][j][1], j]
+            for j in range(len(data[i])):
+                df.loc[len(df)] = [data[i][j][0][0].astype(np.float32), data[i][j][1], data[i][j][0][1]]
 
     progress.update(1)
     progress.set_description("Loaded data into the dataframe")
 
-    return df
+    return df, subdirnames
```

Comment:
Dataset's directories do not need to be returned. The user can obtain them using `os.listdir()` from the `audio_loader`'s data path.

Reply:
@CaptainAlready I agree with this one. We can that way reduce the size of the DF. We could then find the files for manipulation, with the `os.path.join(dataset path, label, filename)`.